src/sentry/plugins/interfaces/releasehook.py

```diff
@@ -27,6 +35,69 @@ def start_release(self, version, **values):
             values=values,
         )
 
+    def set_commits(self, version, commit_list):
```

Comment:
This is going to be potentially painful if this were tons of commits. Not sure how much of an edge case it'll be to have thousands of commits per release. Thinking in the context of maybe an application that's distributed and updated once per month, which has tons of commits between each.

I think we can worry about that later, but it doesn't seem to terrible to remove some O(n) processing here and do bulk `INSERT` queries.